webapp/src/js/components/Map/TableMarkersLayer.js

```diff
@@ -282,6 +319,27 @@ let TableMarkersLayer = React.createClass({
       });
   },
 
+  composeLegendControl(payload) {
+    let {table, markerColourProperty} = payload;
+
+    let legendControl = null;
+
+    if (table !== undefined && markerColourProperty !== undefined) {
+      let legend = ReactDOMServer.renderToStaticMarkup(
```

Comment:
Might be cleaner if the `renderToStaticMarkup` happened on the other side of the add/change call in `Map.js`?